### PR TITLE
refactor(coq): remove coqdep from context

### PR DESCRIPTION
**Pull request by Alizter**
We don't need to carry coqdep around in a record. Since it's program resolution is memoized, we can just grab it when we need it.

I have a bunch of patches removing various things from the context which will allow for further refactoring. i.e. only using what we need to and simplifying the code.

### CODE REVIEW

**Comment by ejgallego** [comment]
> I have a bunch of patches removing various things from the context which will allow for further refactoring. i.e. only using what we need to and simplifying the code.

How does this PR make the code simpler?

TBH I'm not a fan of this kind of code grinding.

File: src/dune_rules/coq_rules.ml [hunk 1]
```diff
@@ -446,12 +447,13 @@ let setup_coqdep_rule ~sctx ~loc (cctx : _ Context.t) ~source_rule coq_module =
     ]
   in
   let stdout_to = Coq_module.dep_file ~obj_dir:cctx.dir coq_module in
+  let* coqdep = Context.coqdep ~sctx cctx in
   (* Coqdep has to be called in the stanza's directory *)
-  Super_context.add_rule ~loc sctx
+  Super_context.add_rule ~loc sctx ~dir
```
Comment (ejgallego) [inline src/dune_rules/coq_rules.ml:452]: This change is unrelated to this PR, right?

Comment (Alizter) [reply src/dune_rules/coq_rules.ml:452]: I've just eta expanded the definition to make it more explicit that's all.

**Comment by Alizter** [comment]
This PR on its own doesn't make code simpler. Exploding the context will.

@ejgallego what do you mean by "code grinding"? I've made a self contained patch that is easy to review?

**Comment by Alizter** [comment]
Closing this as this will come later as part of a bigger PR.
